src/OrchardCore.Modules/OrchardCore.Media/Permissions.cs

```diff
@@ -8,15 +8,20 @@ namespace OrchardCore.Media
     public class Permissions : IPermissionProvider
     {
         public static readonly Permission ManageMedia = new Permission("ManageMediaContent", "Manage Media");
-        public static readonly Permission ManageAttachedMediaFieldsFolder = new Permission("ManageAttachedMediaFieldsFolder", "Manage Attached Media Fields Folder");
+        public static readonly Permission ManageMediaFolder = new Permission("ManageMediaFolder", "Manage All Media Folder");
+        public static readonly Permission ManageOthersMedia = new Permission("ManageOthersMediaContent", "Manage Media For Others", new[] { ManageMediaFolder });
+        public static readonly Permission ManageOwnMedia = new Permission("ManageOwnMediaContent", "Manage Own Media" , new[] { ManageOthersMedia });        
+        public static readonly Permission ManageAttachedMediaFieldsFolder = new Permission("ManageAttachedMediaFieldsFolder", "Manage Attached Media Fields Folder", new[] { ManageMediaFolder});        
```

Comment:
Looking good, but I have some questions around the new permission structure

By this new permission structure `ManageMediaFolder` is now the key permission required to view general media in the media folder (i.e. media that is not Own or Others?)

Is that correct, or have I misread?

If so, `ManageAttachedMediaFieldsFolder` should not be implied by it, as that is a specific admin permission.
Same with `ManageOthersMedia`, should be specific permission.

An Editor and the other users should have permission to view standard files in the media library, by default.

Otherwise the behaviour is going to change significantly for current users.

The key for this pr, is for existing behaviour to remain the same, and to allow consumers to reduce permission, if they wish to restrict users to just their OwnMedia, and make use of the feature.

Reply:
`ManageMediaFolder` is like a super admin permission, can access all the folder under media, include own,  others folder, attachedMediaFields folder. 

The key permission is still `ManageMedia` , Not 'ManageMediaFolder'.
at `ManageMediaFolderAuthorizationHandler.cs`, firstly the permission is set to Permissions.ManageMedia, then the permission morphed according the folder name.

if not folder name(ie. attachedMediaField, own, others) match, that means user request a general media resource and the request permission will be `ManageMedia`. So it will not effect current users, current users should have `ManageMedia` 

let me know if it still not easy to understand, or anything need to improve. Thank you.

Reply:
ok, so that makes sense, but it wasn't obvious.